lib/cloudant.js

```diff
@@ -292,6 +293,25 @@ function findRecursive(mo, query, docs, include, cb) {
   mo.db.find(query, function(err, rst) {
     debug('Cloudant.prototype.all (findRecursive) results: %j %j', err, rst);
     if (err) return cb(err);
+
+    // sort back the ids in "numerical order" if they are numeric
+    query.sort.forEach(function(queries) {
```

Comment:
We only sort them if they give want to sort them in order. i.e if the end user filters it by 
`{order : { id ASC}}` .